src/languages/java.ts

```diff
@@ -66,7 +71,10 @@ const nodeMatchers: Partial<Record<ScopeType, NodeMatcherAlternative>> = {
     "method_declaration.identifier!",
     "constructor_declaration.identifier!",
   ],
-  value: leadingMatcher(["*[declarator][value]", "*[value]"], ["="]),
+  value: leadingMatcher(
+    ["*[declarator][value]", "assignment_expression[right]", "*[value]"],
+    ["="]
```

Comment:
Do we want to add `+=` etc here?

Reply:
On second thought, did you leave it out because it doesn't really make sense? I could see that. Tho I guess you might use it if the increment is a sub expression? That's legal in some languages I think. Eg 

```
if (a += 2) {...}
```

Then you might say "chuck two" and want just the `a`

Reply:
I'm unsure about this. I think that we probably should support identifiers in more types of expression if that's the case

Reply:
I create an issue for it. Will have a proper look when I have time
https://github.com/pokey/cursorless-vscode/issues/533https://github.com/pokey/cursorless-vscode/issues/533